### PR TITLE
ARROW-1246: [Format] Draft Flatbuffer metadata description for Map

### PR DESCRIPTION
**Pull request by wesm**

### CODE REVIEW

**Review by xhochy — APPROVED** [review]
+1, sounds reasonable from my side.

**Comment by BryanCutler** [comment]
Sounds good to me.  I was thinking about how to store Map data from Spark, which is just (key, value) pairs, in Arrow and assumed I'd have to make the types myself into something like `List<entry: Struct<key: K, value: V>>`.  So if I understand this change correctly, when reading back this type of data, I would know from the `Map` type that it was meant to be (key, value) pairs?

File: format/Schema.fbs [hunk 1]
```diff
@@ -44,6 +44,36 @@ table FixedSizeList {
...
+/// has two children: key type and the second the value type. The names of the
+/// child fields may be respectively "entry", "key", and "value", but this is
+/// not enforced
+///
+/// Map
+///   - child[0] entry: Struct
+///     - child[0] key: K
+///     - child[1] value: V
+///
+/// As a matter of convention, neither values of the "entry" nor "key" fields
```
Comment (jacques-n) [inline format/Schema.fbs]: Not sure your meaning here. Seems like this should be a formal specification, not a convention.

**Review by jacques-n — COMMENTED** [review]
Generally looks good. One small question

File: format/Schema.fbs (hunk 1 above)
Comment (wesm) [inline format/Schema.fbs]: Good point; I think this is just a language issue. What I mean is that "neither entry nor key fields may have nulls"

Comment (jacques-n) [inline format/Schema.fbs]: I'm +1 after you clarify. Thanks!

**Comment by wesm** [comment]
Thanks for the review, all. @jacques-n is there specialized code for handling maps in this form in the Arrow Java library, or Dremio?

**Comment by jacques-n** [comment]
No, we should open a jira to support

**Comment by wesm** [comment]
ARROW-1261

**Pull request closed without merging**